server/models/user.ts

```diff
@@ -38,7 +41,7 @@ export class User extends Model<UserAttributes> implements UserAttributes {
   }
```

Comment:
Since you defined UserId in the Post model, for the association you should do:

`User.hasMany(model.Post, { foreignKey: {name: "UserId", allowNull: false } });`